File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.py
class Solution:
    def smallestRepunitDivByK(self, k: int) -> int:
        length = 0
        rem = 0
        for _ in range(k):
            rem = (rem * 10 + 1) % k
            length += 1
            if rem == 0:
                return length
        return -1

File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.cpp
class Solution {
public:
    int smallestRepunitDivByK(int k) {
        if (k == 2 || k == 5) return -1;

        int rem = 0;
        int len = 0;
        vector<bool>visited(k, false);
        while (true) {
            rem = (rem * 10 + 1) % k;
            if (visited[rem]) return - 1;
            visited[rem] = true;
            len++;
            if (rem == 0) return len;
        }
        return -1;
    }
};